app.js

```diff
@@ -1,11 +1,32 @@
 const express = require("express");
 const app = express();
-const { getTopics, getAll } = require("./controller/app.controller");
+const {
+  getTopics,
+  getAll,
+  getArticleById,
+} = require("./controller/app.controller");
 
 app.use(express.json());
 
 app.get("/api/topics", getTopics);
 app.get("/api", getAll);
+app.get("/api/articles/:article_id", getArticleById);
+
+app.use((err, req, res, next) => {
+  switch (err.code) {
+    case "22P02":
+      res.status(400).send({ msg: "Bad request" });
+      break;
+  }
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  }
+  next(err);
+});
```

Comment:
Might be nice to extract this middleware out into another file errors.js or something :) It keeps app.js tidier, see the bottom of the error handling notes for info on this:

https://notes.northcoders.com/courses/js-back-end/error-handling#4-refactoring-to-individual-error-handling-blocks